### ISSUE TITLE
Disabling components

### ISSUE THREAD

**Issue by heuels** [briskhome/briskhome#36]
Disabling/enabling components is yet another feature that should extend component interaction functionality (see #15). One of the possible ways to allow disabling components is writing a `disabled` key to component's `package.json` file like this:

```json
{
  "name": "briskhome-irrigation",
  "version": "0.2.0",
  "main": "irrigation.js",
  "author": "Egor Zaitsev <ezaitsev@briskhome.com>",
  "private": true,
  "plugin": {
    "consumes": [
      "log",
      "db",
      "config",
      "bus"
    ],
    "provides": [
      "irrigation"
    ],
    "disabled": true
  },
  "dependencies": {
    "mongoose": "^4.4.14"
  }
}
```

And then checking it in component loader ([`lib/index.js`](https://github.com/heuels/briskhome/blob/develop/lib/index.js)): 

```js
components = components.concat(fs.readdirSync(directory)
      .filter(subdirectory => directory === './lib'
        ? fs.statSync(path.resolve(directory, subdirectory)).isDirectory()
          && subdirectory.indexOf('core.') === 0
          // the next line checks disabled flag
          && !(require(path.resolve(directory, subdirectory, 'package.json')).plugin.disabled)
        : fs.statSync(path.resolve(directory, subdirectory)).isDirectory()
          && subdirectory.indexOf('briskhome-') === 0)
          // the next line checks disabled flag
          && !(require(path.resolve(directory, subdirectory, 'package.json')).plugin.disabled)
      .map(subdirectory => path.resolve(directory, subdirectory))
);
```

Note that this snippet is not the way it should look in production. The task probably calls for several functions that will filter the main component list and return respectively enabled and disabled components. Of course after disabling/enabling components the application should be restarted. Some kind of solution that would work with both **forever** and **systemd** obviously is required.

**Comment by heuels** [comment]
Moved this to **0.4.0** milestone since it is now a part of a bigger #38 issue which would be a part of the next release.

**Comment by heuels** [comment]
Technically, this issue is resolved as of e1677ee5c96069ccc2665a65cd2be5d45d998c81. Integration with architect (or other custom component loader) is not required.
